pkg/storage/storage.go

```diff
@@ -157,6 +149,50 @@ func (s *Module) dump() {
 
 }
 
+// deviceType gets the device type of a disk
+func (s *Module) deviceType(device filesystem.DeviceInfo, vm bool) (zos.DeviceType, error) {
+	log.Debug().Str("device", device.Path).Msg("checking device type in disk")
+	typ, ok, err := device.Type()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get device type")
+	}
+
+	if ok {
+		log.Debug().Str("device", device.Path).Str("type", typ.String()).Msg("device type loaded from disk")
+		return typ, nil
+	}
+
+	log.Debug().Str("device", device.Path).Msg("checking device type in cache")
+	typ, ok = s.cache.Get(device.Name())
+	if ok {
+		log.Debug().Str("device", device.Path).Str("type", typ.String()).Msg("device type loaded from cache")
+		return typ, nil
```

Comment:
don't return here because this type need to be stored on the disk, if u check my psudo code, I check if NOT cache, i do detect then ALWAYS store the result on disk